### PR TITLE
Simplify scaleline calculation

### LINKED ISSUE #3444: scale line control does not always show
To reproduce go to the scale-line example, choose nautical miles and type map.getView().setZoom(15) into the console.

Initially reported here:

http://stackoverflow.com/questions/29333183/openlayer3-scaleline-does-not-show-nautical

What happens is that i becomes -3 in

```
  var i = 3 * Math.floor(
      Math.log(this.minWidth_ * pointResolution) / Math.log(10));
```

and the div will be hidden. I'm guessing the code needs to be enhanced to show fractional values in this case.

### LINKED ISSUE #4572: Scaleline Not working with us_ft projections..Proj4js sets bad units_ in OL (w/ sample fixes)
With The Below Mapguide WKT Projection put into proj4js the units_ object in openlayers becomes a number instead of ft. same with us feet (why does this happen?)
```
PROJCS["MI83-NIF",GEOGCS["LL83",DATUM["NAD83",SPHEROID["GRS1980",6378137.000,298.25722210]],PRIMEM["Greenwich",0],UNIT["Degree",0.017453292519943295]],PROJECTION["Lambert_Conformal_Conic_2SP"],PARAMETER["false_easting",26246719.160],PARAMETER["false_northing",0.000],PARAMETER["central_meridian",-87.00000000000000],PARAMETER["latitude_of_origin",44.78333333333333],PARAMETER["standard_parallel_1",47.08333333333333],PARAMETER["standard_parallel_2",45.48333333333333],UNIT["Foot_Intnl",0.30480000000000]]
```

This is my workaround CODE for getting Coordinate Systems from Proj4js to be friendly with openlayers Scaleline:
```js
var tempproj = 'EPSG:' + RuntimeMap.CoordinateSystem.EpsgCode;
var csproj = ol.proj.get(tempproj);
if (csproj === undefined) {
	proj4.defs('Custom', RuntimeMap.CoordinateSystem.Wkt);
	var newProj = ol.proj.get('Custom');
	if (['METER', 'METERS', 'M'].indexOf(x.units.toUpperCase()) >= 0) {
		newProj.units_ = "m";
	}
	else if (['FOOT_US', 'FOOT-US', 'FT_US', 'FT-US','US','US_FOOT', 'US-FOOT', 'US_FT', 'US-FT','US'].indexOf(x.units.toUpperCase()) >= 0){
		newProj.units_ = "us-ft";
	}
	else if (['FOOT_INTNL', 'FOOT', 'FT'].indexOf(x.units.toUpperCase()) >= 0) {
		newProj.units_ = "ft";
	}
	else if (['DEGREE', 'DEG', 'D'].indexOf(x.units.toUpperCase()) >= 0) {
		newProj.units_ = "degrees";
	}
	ol.proj.addProjection(newProj);
	display_srs = 'Custom';
}
else {
	display_srs = tempproj
}
```
However in OL_Debug there is a major flaw in logic that means no Coordinate system may use Scaleline unless metric or LL source!!!!
This Logic causes the problem:
```js
  goog.asserts.assert(
      ((units == ol.control.ScaleLineUnits.METRIC ||
        units == ol.control.ScaleLineUnits.IMPERIAL ||
        units == ol.control.ScaleLineUnits.US ||
        units == ol.control.ScaleLineUnits.NAUTICAL) &&
       projectionUnits == ol.proj.Units.METERS) ||
      (units == ol.control.ScaleLineUnits.DEGREES &&
       projectionUnits == ol.proj.Units.DEGREES),
      'Scale line units and projection units should match');
```
I changed this to this code:
```js
  goog.asserts.assert(
      ((units == ol.control.ScaleLineUnits.METRIC ||
        units == ol.control.ScaleLineUnits.IMPERIAL ||
        units == ol.control.ScaleLineUnits.US ||
        units == ol.control.ScaleLineUnits.NAUTICAL) &&
       (projectionUnits == ol.proj.Units.METERS ||
       projectionUnits == ol.proj.Units.FEET ||
       projectionUnits == ol.proj.Units.USFEET ||
       projectionUnits == ol.proj.Units.METERS )) ||
      (units == ol.control.ScaleLineUnits.DEGREES &&
       projectionUnits == ol.proj.Units.DEGREES),
      'Scale line units and projection units should match');
```
Which in reality should just plainly pass through anyways!

I then made a few changes that Should Allow anyone to show Metric/FOOT.USFOOT, and DEGREE scale lines on any Projected units (Haven't tested on degrees yet).
```js
var mpu = ol.proj.METERS_PER_UNIT[projectionUnits];

  var nominalCount = this.minWidth_ * pointResolution;
  //**** CHANGE MADE ****
  if (units != ol.control.ScaleLineUnits.DEGREES) { nominalCount *= mpu;}
  var tempnc = nominalCount * mpu;
  var suffix = '';
  if (units == ol.control.ScaleLineUnits.DEGREES) {
    if (nominalCount < 1 / 60) {
      suffix = '\u2033'; // seconds
      pointResolution *= 3600;
    } else if (nominalCount < 1) {
      suffix = '\u2032'; // minutes
      pointResolution *= 60;
    } else {
      suffix = '\u00b0'; // degrees
    }
  } else if (units == ol.control.ScaleLineUnits.IMPERIAL) {
    if (nominalCount < 0.9144) {
      suffix = 'in';
      pointResolution /= 0.0254;
    } else if (nominalCount < 1609.344) {
      suffix = 'ft';
      pointResolution /= 0.3048;
    } else {
      suffix = 'mi';
      pointResolution /= 1609.344;
    }
  } else if (units == ol.control.ScaleLineUnits.NAUTICAL) {
    pointResolution /= 1852;
    suffix = 'nm';
  } else if (units == ol.control.ScaleLineUnits.METRIC) {
    if (nominalCount < 1) {
      suffix = 'mm';
      pointResolution *= 1000;
    } else if (nominalCount < 1000) {
      suffix = 'm';
    } else {
      suffix = 'km';
      pointResolution /= 1000;
    }
  } else if (units == ol.control.ScaleLineUnits.US) {
    if (nominalCount < 0.9144) {
      suffix = 'in';
      pointResolution *= 39.37;
    } else if (nominalCount < 1609.3472) {
      suffix = 'ft';
      pointResolution /= 0.30480061;
    } else {
      suffix = 'mi';
      pointResolution /= 1609.3472;
    }
  } else {
    goog.asserts.fail('Scale line element cannot be updated');
  }
  
  var i = 3 * Math.floor(
      Math.log(this.minWidth_ * pointResolution) / Math.log(10));
  var count, width;
  //**** CHANGE MADE ****
  if (units != ol.control.ScaleLineUnits.DEGREES) { pointResolution *= mpu; }
  
  while (true) {
    count = ol.control.ScaleLine.LEADING_DIGITS[i % 3] *
        Math.pow(10, Math.floor(i / 3));
    width = Math.round(count / pointResolution);
    if (isNaN(width)) {
      goog.style.setElementShown(this.element_, false);
      this.renderedVisible_ = false;
      return;
    } else if (width >= this.minWidth_) {
      break;
    }
    ++i;
  }
```
Please Let me know if there is any flaw in this logic and whether this can be updated in the future

### PR DESCRIPTION
**Pull request by ahocevar**
It looks like the scaleline control has some legacy code that is no longer needed now that the projection provides a point resolution in crs units.

Fixes #4572.
Fixes #3444.

### CODE REVIEW

**Comment by flipper44** [comment]
capture
This is a picture of a one square mile block and a resulting metric scaleline, after zooming the scaleline crashes

**Comment by ahocevar** [comment]
Could you please provide more context @flipper44, i.e. the code you used to produce the above image?

**Comment by flipper44** [comment]
Its hard to show the Ful;l code because it is dynamically built.
Here is what I found when using this Projection WKT
WKT ="PROJCS[\"IN83-EF\",GEOGCS[\"LL83\",DATUM[\"NAD83\",SPHEROID[\"GRS1980\",6378137.000,298.25722210]],PRIMEM[\"Greenwich\",0],UNIT[\"Degree\",0.017453292519943295]],PROJECTION[\"Transverse_Mercator\"],PARAMETER[\"false_easting\",328083.333],PARAMETER[\"false_northing\",820208.333],PARAMETER[\"scale_factor\",0.999966666667],PARAMETER[\"central_meridian\",-85.66666666666670],PARAMETER[\"latitude_of_origin\",37.50000000000000],UNIT[\"Foot_US\",0.30480060960122]]";
proj4.defs('Custom', Wkt);
var x = proj4.defs('Custom');
var newProj = ol.proj.get('Custom');

Everything works up until my first Zoom (scale Bar values are still wrong) because viewState.Resolution is NaN PS this also causes ZoomSlider to crash

if I set newProj.units_ = "us-ft";
then it works (still bad scale values)
Edit:
I have found that I must do this because the value for that comes from proj4js and OL is "foot_us" and not "us-ft"..This WKT came from a Mapguide Runtimemap

**Comment by flipper44** [comment]
Havent tested it fully but it would seem this line should be changed 
      projection.getPointResolution(viewState.resolution, center) /
      metersPerUnit;
to
      projection.getPointResolution(viewState.resolution, center) *
      metersPerUnit;

**Comment by ahocevar** [comment]
You're right @flipper44, the point resolution needs to be multiplied by the meters per unit. I made that change. Now I'll create an example with your WKT def to see if I can reproduce what else is wrong.

**Comment by ahocevar** [comment]
@flipper44 513cf13 fixes your other issue. I also added an example.

This is now ready for review.

File: examples/scaleline-indiana-east.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    new ol.layer.Tile({
+      source: new ol.source.OSM()
+    })
+  ],
+  target: 'map',
+  view: new ol.View({
+    projection: 'Indiana-East',
+    center: ol.proj.fromLonLat([-85.685, 39.891], 'Indiana-East'),
+    zoom: 7,
+    minZoom: 4
```
Comment (tschaub) [inline examples/scaleline-indiana-east.js]: My computer stops working when I get to 5.  Might be worth limiting this to 6.

Comment (tschaub) [inline examples/scaleline-indiana-east.js]: Could also be that enforcing a center constraint with `extent` would help.

**Comment by tschaub** [comment]
image

Looks like there is an issue when setting `units: 'degrees'`.

Really nice to see the removal of the unnecessary code here.

**Comment by flipper44** [comment]
Re-projection on smaller local coordinate systems does get odd when zOomed way past the local extents.
Also I think there was a reason I had an if statement for degrees in my initial example

**Comment by ahocevar** [comment]
Zoom constraint changed, center constraint introduced, degree issue fixed.

**Comment by fredj** [comment]
LGTM

**Pull request merged**